### PR TITLE
[BUGFIX] Safeguard against using V2 API with V3 Configuration

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Detect Great Expectations configuration version and insure that an error is raised if CLI API version V2 is attempted.
* Refactor `DataContext` initialization to `CLICK.CTX`, which runs at CLI initiation (and remove repetitive code from the implementations of the specific CLI commands).

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-286
-
-

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/cli/cli.py [hunk 1]
```diff
@@ -156,6 +171,14 @@ def cli(ctx, v3_api, verbose, config_file_location, assume_yes):
     else:
         cli_message("Using v2 (Batch Kwargs) API")
 
+        ge_config_version: float = (
+            ctx.obj.get_data_context_from_config_file().get_config().config_version
+        )
+        if ge_config_version >= FIRST_GE_CONFIG_VERSION_WITH_CHECKPOINT_STORE:
+            raise ge_exceptions.InvalidDataContextConfigError(
+                f"Using the legacy v2 (Batch Kwargs) API with a recent config version ({ge_config_version}) is illegal."
+            )
```
Comment (Shinnnyshinshin) [inline great_expectations/cli/cli.py:180]: ❤️

File: great_expectations/cli/cli.py [hunk 2]
```diff
@@ -22,14 +27,24 @@ def __init__(
...
 
+    def get_data_context_from_config_file(self) -> DataContext:
+        directory: str = toolkit.parse_cli_config_file_location(
+            config_file_location=self.config_file_location
+        ).get("directory")
+        context: DataContext = toolkit.load_data_context_with_error_handling(
+            directory=directory,
+            from_cli_upgrade_command=False,
+        )
+        return context
```
Comment (Shinnnyshinshin) [inline great_expectations/cli/cli.py:46]: Love this :)

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM :) Thank you @alexsherstinsky

File: great_expectations/cli/cli.py (hunk 1 above)
Comment (anthonyburdi) [reply great_expectations/cli/cli.py:180]: NICE! Much better error messsage. Thank you.

**Pull request merged**